arelle/ModelValue.py

```diff
@@ -122,80 +124,82 @@ def qnameEltPfxName(element, prefixedName, prefixException=None):
             namespaceURI = None # cancel namespace if it is a zero length string
     return QName(prefix, namespaceURI, localName)
 
+
+def _default_to_empty(string: Optional[str]) -> str:
+    return string or ''
+
+
 class QName:
     __slots__ = ("prefix", "namespaceURI", "localName", "qnameValueHash")
-    def __init__(self,prefix,namespaceURI,localName):
+
+    def __init__(self, prefix: Optional[str], namespaceURI: Optional[str], localName: Optional[str]):
         self.prefix = prefix
         self.namespaceURI = namespaceURI
         self.localName = localName
-        self.qnameValueHash = hash( (namespaceURI, localName) )
+        self.qnameValueHash = hash((namespaceURI, localName))
+
     def __hash__(self):
         return self.qnameValueHash
+
     @property
-    def clarkNotation(self):
+    def clarkNotation(self) -> Optional[str]:
         if self.namespaceURI:
             return '{{{0}}}{1}'.format(self.namespaceURI, self.localName)
         else:
             return self.localName
+
     @property
-    def expandedName(self):
+    def expandedName(self) -> str:
         return '{0}#{1}'.format(self.namespaceURI or "", self.localName)
+
     def __repr__(self):
         return self.__str__() 
+
     def __str__(self):
-        if self.prefix and self.prefix != '':
-            return self.prefix + ':' + self.localName
+        if self.prefix:
+            return '{}:{}'.format(self.prefix, self.localName)
         else:
             return self.localName
-    def __eq__(self,other):
-        try:
-            return (self.qnameValueHash == other.qnameValueHash and 
+
+    def __eq__(self, other):
+        if isinstance(other, QName):
+            return (self.qnameValueHash == other.qnameValueHash and
                     self.localName == other.localName and self.namespaceURI == other.namespaceURI)
```

Comment:
preexisting nit: I'd move each comparison to a separate line for improved readability
```suggestion
            return (self.qnameValueHash == other.qnameValueHash and
                    self.localName == other.localName and
                    self.namespaceURI == other.namespaceURI)
```